pina/model/spline.py

```diff
@@ -1,109 +1,238 @@
+            must contain the keys ``"n"``, ``"min"``, ``"max"``, and ``"mode"``.
+            Here, ``"n"`` specifies the number of knots, ``"min"`` and ``"max"``
+            define the interval, and ``"mode"`` selects the sampling strategy.
+            The supported modes are ``"uniform"``, where the knots are evenly
+            spaced over :math:`[min, max]`, and ``"auto"``, where knots are
+            constructed to ensure that the spline interpolates the first and
+            last control points. In this case, the number of knots is adjusted
+            if :math:`n < 2 * order`. If None is given, knots are initialized
+            automatically over :math:`[0, 1]` ensuring interpolation of the
+            first and last control points. Default is None.
+        :type knots: torch.Tensor | dict
+        :param torch.Tensor control_points: The control points of the spline.
+            If None, they are initialized as learnable parameters with an
+            initial value of zero. Default is None.
+        :raises AssertionError: If ``order`` is not a positive integer.
+        :raises ValueError: If both ``knots`` and ``control_points`` are None.
+        :raises ValueError: If ``knots`` is not one-dimensional.
+        :raises ValueError: If ``control_points`` is not one-dimensional.
+        :raises ValueError: If the number of ``knots`` is not equal to the sum
+            of ``order`` and the number of ``control_points.``
+        :raises UserWarning: If the number of control points is lower than the
+            order, resulting in a degenerate spline.
         """
         super().__init__()
 
-        check_consistency(order, int)
+        # Check consistency
+        check_positive_integer(value=order, strict=True)
 
-        if order < 0:
-            raise ValueError("Spline order cannot be negative.")
+        # Raise error if neither knots nor control points are provided
         if knots is None and control_points is None:
-            raise ValueError("Knots and control points cannot be both None.")
+            raise ValueError("knots and control_points cannot both be None.")
 
-        self.order = order
-        self.k = order - 1
-
-        if knots is not None and control_points is not None:
-            self.knots = knots
-            self.control_points = control_points
+        # Initialize knots if not provided
+        if knots is None and control_points is not None:
+            knots = {
+                "n": len(control_points) + order,
+                "min": 0,
+                "max": 1,
+                "mode": "auto",
+            }
 
-        elif knots is not None:
-            print("Warning: control points will be initialized automatically.")
-            print("         experimental feature")
+        # Initialization - knots and control points managed by their setters
+        self.order = order
+        self.knots = knots
+        self.control_points = control_points
+
```

Comment:
Just a question, why we inizialize control points as none? It seems you always have to pass the control points. In this case let's remove the default none